### ISSUE TITLE
Unable to read old RPMs with 4.16

### ISSUE THREAD

**Issue by jengelh** [rpm-software-management/rpm#1762]
The current rpm version as shipped by openSUSE Tumbleweed (I hope it's not a patch of the distro):

```
» wget https://ftp.gwdg.de/pub/linux/suse/5.3/i386.de/suse/a1/aaa_base.rpm
» rpm -q rpm
rpm-4.16.1.3-3.1.x86_64
» rpm -qp aaa_base.rpm
error: aaa_base.rpm: invalid signature tag Archivesize (1046)
error: aaa_base.rpm: not an rpm package (or package manifest)
```

Meanwhile, the slightly older rpm from openSUSE Leap 15.2:

```
» rpm -q rpm
rpm-4.14.1-lp152.17.5.x86_64
» rpm -qp aaa_base.rpm 
aaa_base-98.8.12-0.i386
```

**Comment by pmatilai** [comment]
Already fixed upstream but pending an update to the 4.16 branch: https://github.com/rpm-software-management/rpm/commit/e2f1f1931c5ccf3ecbe4e1e12cacb1e17a277776

**Comment by dmnks** [comment]
We never did an update on the 4.16 branch (4.16.2) and that release is now out of support. Closing.
